cukinia

```diff
@@ -170,24 +166,21 @@ _cukinia_user()
 # cukinia_process: checks if process $pname runs
 # Optional: check if the process exist for a given username $puser
 # arg: $1: the process name
-# arg: $2: the user name
+# arg: $2: [optional] the user name
 _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
 
-	if [ -n "$puser" ]; then
-		_cukinia_prepare "Checking process \"$pname\" with user $puser"
-		local result=$(ps -a -o user,comm | awk ' $1 == "'$puser'" && $2 == "'$pname'" { print $2 } ')
-	else
-		_cukinia_prepare "Checking process \"$pname\""
-		local result=$(ps -a -o comm | awk ' $1 == "'$pname'" { print $1 } ')
-	fi
-	[ -n "$result" ] && return 0 || return 1
+	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
+	local ret=$(ps -a -o user,comm |
+		      awk '$2 == "'$pname'" && $1 ~ /^'${puser:-.*}'$/ { print }')
```

Comment:
I tend to favor readability over line numbers, esp. given they are two distincts programs (ps and awk). I hesitated to breaking down the awk code itself even more to make it even easier to read.
https://paste.fedoraproject.org/paste/aG5sarpTXhRC7xpyFHrHgl5M1UNdIGYhyRLivL9gydE=